ext/openssl/openssl.c

```diff
@@ -3709,8 +3710,7 @@ static EVP_PKEY * php_openssl_generate_private_key(struct php_x509_request * req
 	EVP_PKEY * return_val = NULL;
 
 	if (req->priv_key_bits < MIN_KEY_LENGTH) {
-		php_error_docref(NULL, E_WARNING, "Private key length is too short; it needs to be at least %d bits, not %d",
-				MIN_KEY_LENGTH, req->priv_key_bits);
+		zend_value_error("Private key must be at least %d bits", MIN_KEY_LENGTH);
```

Comment:
this will be inconsistent with other checks - please change to warning.

Reply:
We changed a bunch of minimal values to use ValueError in other extensions while still keeping E_WARNINGs for other cases. Same goes for the empty case for strings/arrays and I don't exactly see why OpenSSL should get a pass on this to keep everything as warnings.
ValueErrors are there to point out programming errors, I understand why the signature or the coercion ones need to be kept as Warnings as they might be due to the OpenSSL version/file environment, but here I don't see why it should be kept a warning.

Reply:
Well this comes from openssl configuration (e.g. openssl.cnf or system config) so it's not really a programming error but more configuration error. Basically it's not like user passed different value.